src/cli_parser.rs

```diff
@@ -0,0 +1,184 @@
+        // Exclude pattern in next element
+        Option::None => return (index, Result::Ok(CLIOptions::End)),
+        Option::Some(arg0) => match arg0.as_str().trim() {
+            // Exclude String:
+            "-e" => match args.get(index + 1) {
+                Option::None => {
+                    return (
+                        index,
+                        Result::Err(String::from("missing Argument after -e")),
+                    )
+                }
+                Option::Some(arg1) => {
+                    return (
+                        index + 2,
+                        Result::Ok(CLIOptions::ExcludeString {
+                            string: String::from(arg1),
+                        }),
+                    )
+                }
+            },
+            //Exclude regex
+            "-er" => match args.get(index + 1) {
+                Option::None => {
+                    return (
+                        index,
+                        Result::Err(String::from("missing Argument after -er")),
+                    )
+                }
+                Option::Some(arg1) => {
+                    return (
+                        index + 2,
+                        Result::Ok(CLIOptions::ExcludeRegex {
+                            regex: String::from(arg1),
+                        }),
+                    )
+                }
+            },
+            // help tag
+            "-?" => return (index + 1, Result::Ok(CLIOptions::Help)),
+            // verbose tag
+            "-v" => return (index + 1, Result::Ok(CLIOptions::Verbose)),
+            // other: source + destination
+            _ => match args.get(index + 1) {
+                Option::None => return (index, Result::Err(String::from("no destination given"))),
+                Option::Some(arg1) => {
+                    return (
+                        index + 2,
+                        Result::Ok(CLIOptions::SourceDestination {
+                            source: PathBuf::from(arg0),
+                            destination: PathBuf::from(arg1),
+                        }),
+                    )
+                }
+            },
+        },
+    }
+}
+
+fn print_error(index: &usize, msg: String) {
+    println!("Error at option at index {}, message: \"{}\"", index, msg);
+    println!("use \"-?\" for help");
+}
+
+fn print_help_message() {
+    // Help message:
+    println!(
+        r#"
+    usage: backup-tool [options] <source> <destination>
+        <source>        : Sourcepath for the backup
+        <destination>   : Destination path for the backup
+        options :
+            -e          : Folder or File to exclude (can be given more than once)
+            -er         : Regular expression to exclude (can be given more than once)
+            -v          : verbose output
+            -?          : help
+    "#
+    );
+    // End Help message
+}
+
+pub fn parse_options(args: Vec<String>) -> configuration::ConfStruct {
```

Comment:
The function could return `Result<ConfStruct>`. That way we can avoid the panic! at line 164